pybamm/solvers/idaklu_solver.py

```diff
@@ -251,6 +275,49 @@ def resfn(t, y, inputs, ydot):
                 "mass_action", [v_casadi], [casadi.densify(mass_matrix @ v_casadi)]
             )
 
+            # if output_variables specified then convert functions to casadi
+            # expressions for evaluation within the idaklu solver
+            self.var_casadi_fcns = {}
```

Comment:
all the expressions are typically converted to casadi function within the base solver (in `set_up`). Probably best to do this there so that other solvers can use them

Reply:
I've now moved the generation of casadi variable (and sensitivity) functions to within the base_solver (with some renaming for clarity), while retaining the idaklu conversion code within the idaklu_solver file.